packages/headless/src/controllers/commerce/product-listing/headless-product-listing.ts

```diff
@@ -0,0 +1,132 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {CommerceAPIErrorStatusResponse} from '../../../api/commerce/commerce-api-error-response';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {configuration} from '../../../app/common-reducers';
+import {ProductRecommendation} from '../../../commerce.index';
+import {
+  fetchProductListing,
+  setProductListingUrl,
+} from '../../../features/product-listing/v2/product-listing-v2-actions';
+import {productListingV2Reducer as productListing} from '../../../features/product-listing/v2/product-listing-v2-slice';
+import {loadReducerError} from '../../../utils/errors';
+import {validateOptions} from '../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+const optionsSchema = new Schema({
+  url: new StringValue({
+    required: true,
+    url: true,
+  }),
+});
+
+export interface ProductListingOptions {
```

Comment:
I think it would make more sense to have context labels set in the context controller. What do you think? 

It feels strange to have the context labels set through the product listings controller.

Reply:
I might be misunderstanding what labels are intended for, but I thought they were specific to each product listing page, i.e., not "global" context that would be relevant to an entire commerce app.

If my understanding is correct, then I think labels should be on the product listing controller. Otherwise, they should be on the context controller.

Reply:
If labels are indeed something that are "unique per product listing page", and that you don't need to have any impact outside of the product list per say (ie: if you have labels X for a specific product listing page, it has no impact on the cart), then they are "local" context that need to be kept local (ie: only on the product listing controller).

I guess, it would help if you could you explain what label are for ?

Reply:
Ok, after reading the RFC about what they represent, its definitely something "local" to a specific page, not global to the full website.

As such, it looks like what Fred is suggesting is the right approach.

Reply:
After reading the RFC myself, I tend to agree with @samisayegh 's comment here: https://docs.google.com/document/d/1HIlPbSVm9luLfObJvoIyCvTpE37O-ts-bQGO8Np5cKo/edit?disco=AAAA4HbWQX4

I.e., it also seems to me like labels are a pre-emptive escape hatch. This leads me to believe that we should not expose this parameter in Headless out of the gate after all. We can wait until someone actually needs it, which might never happen for all we know.

Reply:
I'll resolve this conversation. Let's add support for context labels as the need comes up.

Thanks y'all!